### PR TITLE
Retry a few times before killing docker daemon

### PR DESCRIPTION
**Pull request by honnix**
**What this PR does / why we need it**:
In our k8s setup, sometime docker becomes very unresponsive, and recovers after a while. So timeout after 60s doesn't help much. We retry a few times before killing docker daemon.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cluster/gce/gci/health-monitor.sh [hunk 1]
```diff
@@ -26,7 +26,19 @@ set -o pipefail
 # automatically restart the process.
 function docker_monitoring {
   while [ 1 ]; do
-    if ! timeout 60 docker ps > /dev/null; then
+    n=0
+    dockerpscheck=1
+    until [ $n -ge 10 ]; do
+      if ! timeout 60 docker ps > /dev/null; then
```
Comment (honnix) [inline cluster/gce/gci/health-monitor.sh]: We can probably tweak the retry times. We are not entirely sure how many retries it happened in our environment, but simply waiting for 60s has made docker being killed innocently.

`docker ps` becomes very unresponsive under certain conditions. This PR surely is merely a workaround.

Comment (honnix) [inline cluster/gce/gci/health-monitor.sh]: Also with this fix we haven't seen docker daemon being killed "accidentally".

**Comment by honnix** [comment]
I signed it!

**Comment by thkoch2001** [comment]
honnix, I also believe that this PR does fit for your particular use case. But in general your fix would mean that new pods would not be scheduled on a node for 10 minutes when docker gets stuck and a simple restart of the docker daemon would "fix" this.

I suggest to think about other ways to fix the problem of docker getting stuck:

- Use rkt as an alternative runtime. It does not need a daemon and such it also can not get stuck like docker.
- Monitor the responsiveness of the docker daemon and stop all requests to the docker daemon already when the docker daemon gets slightly loaded. So this should hopefully prevent it from getting stuck in the first place?
- Work with docker project to identify the root cause of the docker stuck problem and fix it.

**Comment by dchen1107** [comment]
@honnix I am going to close this pr due to inactivity. I believe I shared the same concern as @thkoch2001 with you in another thread. If you disagreed, please reopen it. We can discuss more.

**Pull request closed without merging**